backend/migrations/versions/852cdf6cf1e0_add_redshift_datasets.py

```diff
@@ -0,0 +1,130 @@
+
+
+def upgrade():
+    op.create_table(
+        'redshift_connection',
+        sa.Column('label', sa.String(), nullable=False),
+        sa.Column('name', sa.String(), nullable=False),
+        sa.Column('owner', sa.String(), nullable=False),
+        sa.Column('created', sa.DateTime(), nullable=True),
+        sa.Column('updated', sa.DateTime(), nullable=True),
+        sa.Column('deleted', sa.DateTime(), nullable=True),
+        sa.Column('description', sa.String(), nullable=True),
+        sa.Column('tags', postgresql.ARRAY(sa.String()), nullable=True),
+        sa.Column('connectionUri', sa.String(), nullable=False),
+        sa.Column('environmentUri', sa.String(), nullable=False),
+        sa.Column('SamlGroupName', sa.String(), nullable=False),
+        sa.Column('redshiftType', sa.String(), nullable=False),
+        sa.Column('clusterId', sa.String(), nullable=True),
+        sa.Column('nameSpaceId', sa.String(), nullable=True),
+        sa.Column('workgroup', sa.String(), nullable=True),
+        sa.Column('database', sa.String(), nullable=False),
+        sa.Column('redshiftUser', sa.String(), nullable=True),
+        sa.Column('secretArn', sa.String(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ['environmentUri'],
+            ['environment.environmentUri'],
+        ),
+        sa.PrimaryKeyConstraint('connectionUri'),
+    )
+
+    op.create_table(
+        'redshift_dataset',
+        sa.Column('datasetUri', sa.String(), nullable=False),
+        sa.Column('connectionUri', sa.String(), nullable=False),
+        sa.Column('schema', sa.String(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ['connectionUri'],
+            ['redshift_connection.connectionUri'],
+        ),
+        sa.ForeignKeyConstraint(
+            ['datasetUri'],
+            ['dataset.datasetUri'],
+        ),
+        sa.PrimaryKeyConstraint('datasetUri'),
+    )
+
+    op.create_table(
+        'redshift_table',
+        sa.Column('label', sa.String(), nullable=False),
+        sa.Column('name', sa.String(), nullable=False),
+        sa.Column('owner', sa.String(), nullable=False),
+        sa.Column('created', sa.DateTime(), nullable=True),
+        sa.Column('updated', sa.DateTime(), nullable=True),
+        sa.Column('deleted', sa.DateTime(), nullable=True),
+        sa.Column('description', sa.String(), nullable=True),
+        sa.Column('tags', postgresql.ARRAY(sa.String()), nullable=True),
+        sa.Column('datasetUri', sa.String(), nullable=False),
+        sa.Column('rsTableUri', sa.String(), nullable=False),
+        sa.Column('topics', postgresql.ARRAY(sa.String()), nullable=True),
+        sa.ForeignKeyConstraint(['datasetUri'], ['redshift_dataset.datasetUri'], ondelete='CASCADE'),
+        sa.PrimaryKeyConstraint('rsTableUri'),
+    )
+    op.execute("ALTER TYPE datasettypes ADD VALUE 'Redshift'")
```

Comment:
did we look into if this fails if datasettype of 'Redshift' already exists?

Reply:
I just checked and it fails

Reply:
Added a check to pass the exception if the enum value already exists. That was tested in a local db